### PR TITLE
feat: import v1 android BLE driver

### PR DESCRIPTION
**Pull request by D4ryl00**
- [x] import the BLE driver from Berty v1, adapt it with the driver interface of Berty v2
- [x] move the data cache from drivers to the proximity transport for data packets received before the end of the handshake
- [x] stabilize the driver
- [ ] stabilize Samsung connections

# Help to debug:
 * Disable WIFi/4g on at least one phone
 * List your device IDs:
`adb devices`
* Get logs for each devices:
`adb -s ${DEVICE_ID} logcat "*:V" | grep --line-buffered -E "GoLog|BluetoothGatt|bt_.*|bty\.(ble|ProximityTransport).*:" > a.out`
* Generate Berty connections (contact request, send messages, etc)
* Share your log outputs (a.out, etc). These logs also expose your peer ID and group IDs.

Signed-off-by: D4ryl00 <d4ryl00@gmail.com>

### CODE REVIEW

**Review by clegirar — APPROVED** [review]
LGTM 👍

File: js/android/app/src/main/java/tech/berty/gobridge/bledriver/Scanner.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import android.bluetooth.le.ScanSettings;
+import android.content.Context;
+import android.util.Log;
+
+import java.util.Collections;
+import java.util.List;
+
+// see https://stackoverflow.com/questions/27040086/onbatchscanresults-is-not-called-in-android-ble
+
+// API level 21
```
Comment (aeddi) [inline js/android/app/src/main/java/tech/berty/gobridge/bledriver/Scanner.java:18]: 23?

File: js/go.mod [hunk 2]
```diff
@@ -3,3 +3,5 @@
 module berty.tech/berty/js/v2
 
 go 1.14
+
+require golang.org/x/mobile v0.0.0-20200801112145-973feb4309de // indirect
```
Comment (aeddi) [inline js/go.mod:7]: This file shouldn't contain any mod

**Review by aeddi — APPROVED** [review]
LGTM 👍

**Pull request merged**